### PR TITLE
Make OSDisk. DiskSizeGB optional

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**
/kind feature
/kind api-change

**What this PR does / why we need it**:
Change the field `OSDisk.DiskSizeGB` to a pointer so its optional

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
related to  #618 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Change OSDisk.DiskSizeGB to be optional
```

### CODE REVIEW

**Review by shysank — COMMENTED** [review]
/lgtm with one small clarification.

File: api/v1alpha4/types.go [hunk 1]
```diff
@@ -367,8 +367,11 @@ const (
 // conversion-gen where the warning message generated uses a relative directory import rather than the fully
 // qualified import when generating outside of the GOPATH.
 type OSDisk struct {
-	OSType     string `json:"osType"`
-	DiskSizeGB int32  `json:"diskSizeGB"`
+	OSType string `json:"osType"`
+	// DiskSizeGB is the size in GB to assign to the OS disk.
+	// Will have a default of 30GB if not provided
```
[the remark this answers is not in the document]
Comment (nader-ziada) [reply api/v1alpha4/types.go:372]: its is defaulted by Azure itself if left empty

**Comment by nader-ziada** [comment]
looks like a flake

/test pull-cluster-api-provider-azure-e2e

**Pull request merged**